code/obj/item/mob_parts/robot_parts.dm

```diff
@@ -169,13 +172,46 @@ ABSTRACT_TYPE(/obj/item/parts/robot_parts)
 	proc/on_life()
 		return
 
+	proc/reinforce(var/obj/item/sheet/M, var/mob/user, var/obj/item/parts/robot_parts/result, var/need_reinforced)
+		if (need_reinforced && !M.reinforcement)
+			boutput(user, SPAN_ALERT("You'll need reinforced sheets to reinforce this component."))
+			return
+		if (M.amount < 2)
+			boutput(user, SPAN_ALERT("You need at least two metal sheets to reinforce this component."))
+			return
+		if (M.material != src.material)
+			boutput(user, SPAN_ALERT("You need the same material as the component to reinforce."))
+			return
+
+		var/obj/item/parts/robot_parts/newitem = new result(get_turf(src))
+		newitem.setMaterial(src.material)
+
+		boutput(user, SPAN_NOTICE("You reinforce [src.name] with the metal."))
+		M.change_stack_amount(-2)
+		if (M.amount < 1)
+			user.drop_item()
+			qdel(M)
+
+		if (istype(src,/obj/item/parts/robot_parts/head) && istype(newitem,/obj/item/parts/robot_parts/head))
```

Comment:
`istype(src, ...` is an anti-pattern, consider just using inheritance instead (ie defining this behaviour in an override on the type you want)